### PR TITLE
Feature: Integrate external

### LINKED ISSUE #89: Integrate externalId
## Description

Integrate externalId, externalId is compose to `${type}_${externalId}`

Modify Parser, Render and Component id to retrieve and use external id.

```tf
resource "type" "external_id" {}
resource "aws_vpc" "vpc1" {}
```

### PR DESCRIPTION
**Pull request by aboubakar14**
resolve #89

### CODE REVIEW

File: src/parser/TerraformListener.js [hunk 1]
```diff
@@ -23,19 +21,26 @@ class TerraformListener extends antlr4.tree.ParseTreeListener {
...
+    // key -> type.externalId, value -> id
+    this.ids = new Map();
+    // We need this idCounter because we can't use pluginData to generate the id.
+    this.idCounter = 1;
   }
 
   addComponent() {
     this.currentComponent.path = this.currentFile.path;
-    this.components.push(this.currentComponent);
+    const typeExternalId = `${this.currentComponent.definition.type}.${this.currentComponent.externalId}`;
```
Comment (Zorin95670) [inline src/parser/TerraformListener.js:33]: put fist and add a line after

File: src/parser/TerraformListener.js [hunk 2]
```diff
@@ -217,6 +220,11 @@ class TerraformListener extends antlr4.tree.ParseTreeListener {
       blockType: this.currentBlockType,
       type,
     });
+    if (this.currentBlockType === 'provider') {
```
Comment (Zorin95670) [inline src/parser/TerraformListener.js:223]: break line before

File: src/parser/TerraformListener.js [hunk 3]
```diff
@@ -217,6 +220,11 @@ class TerraformListener extends antlr4.tree.ParseTreeListener {
...
+      this.currentComponent.externalId = type;
+      const typeExternalId = `${type}.${this.currentComponent.externalId}`;
```
Comment (Zorin95670) [inline src/parser/TerraformListener.js:225]: put first and break line after and use type.type and add a comment

File: tests/unit/render/TerraformRenderer.spec.js [hunk 4]
```diff
@@ -338,12 +338,12 @@ describe('Test TerraformRenderer', () => {
         expect(new TerraformRender(metadata.pluginData).renderFiles()).toEqual([input]);
       });
 
-      it('Should render main file', () => {
+      it.only('Should render main file', () => {
```
Comment (justinehell) [inline tests/unit/render/TerraformRenderer.spec.js:341]: remove .only

**Review by justinehell — CHANGES_REQUESTED** [review]

File: tests/unit/render/TerraformRenderer.spec.js (hunk 4 above)
Comment (Zorin95670) [inline tests/unit/render/TerraformRenderer.spec.js:341]: image

Comment (Zorin95670) [reply tests/unit/render/TerraformRenderer.spec.js:341]: https://youtu.be/Ao3vOCDZd-Y?t=6

**Review by justinehell — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**